### DISCUSSION TITLE
Cells with no neighbors

### DISCUSSION THREAD

**Discussion by shahrozeabbas** [stuart-lab/signac#1548] [General]
Hoping to find someone else having a similar problem....has anyone encountered scATAC datasets where after calculating the neighborhood graph, you find that some cells have no neighbors? If so, are there reasons or solutions to this? I suspect that it may have to do with the features/number of dimensions used for neighborhood graph construction.

**Comment by timoast** [comment]
We use a k-nearest neighbor algorithm, so each cell will have k neighbors by definition

**Reply by shahrozeabbas** [reply]
Yeah that's what I expected, that's why I've been confused recently. 

I had to switch over to using `scanpy` because the size of the dataset is over a million cells. The `neighbors` method relies on UMAP to calculate neighbors, but the idea should still be the same. It just doesn't make sense.

**Reply by shahrozeabbas** [reply]
I am performing the TFIDF normalization and opted to subset features per dataset by using the `FindVariableFeatures` function to reduce the size of the dataset after merging. I then take the union across all datasets. You think that may have something to do with it?

**Reply by timoast** [reply]
Typically we do not run `FindVariableFeatures` on scATAC-seq data as the mean and variance are very low across all peaks. You can run `FindTopFeatures` to select features based on total number of counts instead

**Reply by shahrozeabbas** [reply]
It turns out this was the reason for the problem I was seeing! Good to know for anyone else who might experience it as well.

**Comment by shahrozeabbas** [comment]
Hi Tim,

Any thoughts on [poisson gene selection from scvi](https://docs.scvi-tools.org/en/stable/api/reference/scvi.data.poisson_gene_selection.html) for selecting top features in ATAC?

Thanks!
